tests/mintOsToken.test.ts

```diff
@@ -0,0 +1,105 @@
+import {
+    Hex,
+    PrivateKeyAccount,
+    PublicClient,
+    WalletClient,
+    createPublicClient,
+    createWalletClient,
+    http,
+    parseEther,
+    publicActions,
+} from 'viem';
+import { Networks } from '../src/types/enums';
+import { OpusPool } from '../src';
+import { getStakeBalance } from '../src/internal/osTokenRequests';
+import { hardhat } from 'viem/chains';
+import { privateKeyToAccount } from 'viem/accounts';
+import { mine } from '@nomicfoundation/hardhat-toolbox/network-helpers';
+import { expect, test } from '@jest/globals';
+import { VaultABI } from '../src/internal/contracts/vaultAbi';
+
+const VAULT_ADDRESS: Hex = '0x95d0db03d59658e1af0d977ecfe142f178930ac5';
+const AMOUNT_TO_STAKE = parseEther('20');
+const AMOUNT_TO_MINT = 2n;
+
+describe('Minting OsToken', () => {
+    let USER_ADDRESS: Hex;
+    let publicClient: PublicClient;
+    let client: WalletClient;
+    let account: PrivateKeyAccount;
+    beforeAll(() => {
+        USER_ADDRESS = '0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266';
+        publicClient = createPublicClient({
+            chain: hardhat,
+            transport: http(),
+        });
+        account = privateKeyToAccount('0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80');
+        client = createWalletClient({
+            account,
+            chain: hardhat,
+            transport: http(),
+        }).extend(publicActions);
+    });
+    test('should mint OsToken - happy path', async () => {
+        const pool = new OpusPool({
+            address: USER_ADDRESS,
+            network: Networks.Hardhat,
+        });
+
+        const userSharesInitial: bigint = await publicClient.readContract({
+            abi: VaultABI,
+            address: VAULT_ADDRESS,
+            functionName: 'getShares',
+            args: [USER_ADDRESS],
+        });
+
+        // // we stake first
+        const stakeTransactionData = await pool.buildStakeTransaction({
+            vault: VAULT_ADDRESS,
+            amount: AMOUNT_TO_STAKE,
+        });
+
+        const stakeTxHash = await client.sendTransaction({
+            account,
+            to: VAULT_ADDRESS,
+            data: stakeTransactionData.transaction,
+            value: stakeTransactionData.amount,
+            type: 'eip1559',
+            gas: stakeTransactionData.gasEstimation,
+            maxPriorityFeePerGas: stakeTransactionData.maxPriorityFeePerGas,
+            maxFeePerGas: stakeTransactionData.maxFeePerGas,
+            chain: hardhat,
+        });
+        await mine(10);
+        const stakeReceipt = await publicClient.waitForTransactionReceipt({ hash: stakeTxHash });
+        expect(stakeReceipt.status).toEqual('success');
+
+        const stake = await getStakeBalance(pool, VAULT_ADDRESS);
```

Comment:
This should be using `pool.getStakeBalanceForUser`

Reply:
fixed, ty!